Updated header (Nav Bar) for hiding reports page

courtneyc1 opened the pull request:
Hiding Reports page 
due to no current data being presented needed to hide this page until it gets updated. 
frontend/src/components/Header.tsx
commented out line 250-255 to hide page.

# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

frontend/src/components/Header.tsx
comment out line 250-255 to hide the page.

Will need to uncomment when reports page is finished.

## 💭 Motivation and context ##

<!-- Why is this change required? -->
Clients get confused on reports page when nothing is presented currently. 
<!-- What problem does this change solve? How did you solve it? -->
gives dev team a chance to update the reports page

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
used local crossed to verify changes 

![Screenshot 2023-09-13 at 9 03 35 AM](https://github.com/cisagov/crossfeed/assets/126171978/5e9068bc-faa2-4e09-8a36-02c7a93d1fb0)

before when reports page was present to

![Screenshot 2023-09-13 at 9 35 24 AM](https://github.com/cisagov/crossfeed/assets/126171978/5e0ae848-9bab-4d72-beb9-74b6619409b0)


reports page hide on the nav bar.

Matthew-Grayson commented:
@epicfaace Rachel asked Courtney to hide this page until it is functional. I'd like your input as to whether we should comment it out or remove it (based on how we typically handle unused code).

epicfaace reviewed (commented):
I would check with P&E team / Andy / Lamar before removing it. I know the page does include reports for federal agencies enrolled in P&E. If we need to use it for P&E, I’d suggest showing it only for govt agencies organizations and hiding it elsewhere. 

@aloftus23 @stewartl97

epicfaace commented:
> I would check with P&E team / Andy / Lamar before removing it. I know the page does include reports for federal agencies enrolled in P&E. If we need to use it for P&E, I’d suggest showing it only for govt agencies organizations and hiding it elsewhere.
> 
> @aloftus23 @stewartl97

(The reports are only visible if you select a fed agency in the organization dropdown).

epicfaace commented:
> We spoke about it in the meeting today. It looks like it isn't being used on either site at the moment. The consensus was commenting out the code until we add reports functionality.

Yes, in that case, let's comment it out. Good to comment it out as we might need to re-add it in the future.

epicfaace commented on frontend/src/components/Header.tsx:
```diff
@@ -247,12 +247,16 @@ const HeaderNoCtx: React.FC<ContextType> = (props) => {
     { title: 'Feeds', path: '/feeds', users: ALL_USERS, exact: false },
+
+    /* 
+    hiding reprts page until finished 
```
```suggestion
    Hiding Reports page until finished 
```

Matthew-Grayson commented:
I fixed the linting errors and replaced the header snapshot (the snapshot caused failure of the "Header matches snapshot" test. Additionally I fixed a type error related to the Organization interface used for testing by adding pendingDomains to testOrganization.

courtneyc1 commented:
updated typo/ commit from Ashwin's suggestion. Updated the branch since master had some changes since then. All changes approve and branch has no conflicts. Merged to master.
